d3a_interface/sim_results/device_statistics.py

```diff
@@ -136,46 +136,36 @@ def calculate_stats_for_infinite_bus(cls, area_dict, subdict, core_stats, curren
             {current_market_slot: bought_traded_energy})
         cls._calc_min_max_from_sim_dict(subdict, bought_key_name)
 
-    @classmethod
-    def _pv_production_stats(cls, area_dict: Dict, subdict: Dict, core_stats=None,
-                             current_market_slot=None):
-        if core_stats is None:
-            core_stats = {}
-        key_name = "pv_production_kWh"
-        if core_stats[area_dict['uuid']] == {}:
-            return
-
-        create_or_update_subdict(
-            subdict, key_name,
-            {current_market_slot: core_stats[area_dict["uuid"]][key_name]})
-
-        cls._calc_min_max_from_sim_dict(subdict, key_name)
+    @staticmethod
+    def _compute_key_name_for_node(area: Dict):
+        if is_pv_node_type(area):
+            return "pv_production_kWh"
+        if is_prosumer_node_type(area):
+            return "soc_history_%"
+        if is_load_node_type(area):
+            return "load_profile_kWh"
+        if area["type"] == "HomeMeterStrategy":
+            return "home_meter_profile_kWh"
+        if area["type"] == "FinitePowerPlant":
```

Comment:
There is a function that checks this `is_finite_power_plant_node_type` in `d3a_interface/sim_results/__init__.py:36`
I suggest adding a similar function for the HomeMeter + adding it in the `get_unified_area_type` function residing in the same file

Reply:
Plus, this function can be replaced with a mapping dict of `unified_area_type` as keys and corresponding `key_name_for_node` as values
This way the usage would be 
```python
key_name = mapping_dict[area_dict["type"]]
```
However if you think that in the future we need to add more logic here, feel free to keep it

Reply:
I wanted to add a dict or a class to do the mapping, I think it's a good idea. I decided to leave it out of this PR to avoid too much refactoring, since the PR is already big. I can make a new PR and point to this one, so that we can keep things separate.